src/core/task/Task.ts

```diff
@@ -1461,14 +1493,36 @@ export class Task extends EventEmitter<ClineEvents> {
 	}
 
 	public async *attemptApiRequest(retryAttempt: number = 0): ApiStream {
+		const state = await this.providerRef.deref()?.getState()
 		const {
 			apiConfiguration,
 			autoApprovalEnabled,
 			alwaysApproveResubmit,
 			requestDelaySeconds,
 			experiments,
 			autoCondenseContextPercent = 100,
```

Comment:
Only tiny nitpick is that I think you stopped using some of these and are referencing through state?. now. I don't feel strongly which way we go, but probably should pick one.